lib/src/blaze/fetch_full_transaction.rs

```diff
@@ -577,62 +464,163 @@ impl FetchFullTxns {
             wallet_transactions,
             is_outgoing_transaction,
             outgoing_metadatas,
-            Keys::okeys,
-            |t: &Transaction| t.orchard_bundle(),
-            orchard::Bundle::actions,
-            |key| OrchardIvk::try_from(&key.key),
-            OrchardDomain::for_action,
-            <[u8; 512]>::as_slice,
         )
         .await;
     }
 }
 
-async fn scan_bundle<K, B, L, O, D, E>(
-    _config: &ZingoConfig,
+async fn scan_bundle<D>(
+    config: &ZingoConfig,
     transaction: &Transaction,
-    _height: BlockHeight,
-    _unconfirmed: bool,
-    _block_time: u32,
+    height: BlockHeight,
+    unconfirmed: bool,
+    block_time: u32,
     keys: &Arc<RwLock<Keys>>,
-    _wallet_transactions: &Arc<RwLock<WalletTxns>>,
-    _is_outgoing_transaction: &mut bool,
-    _outgoing_metadatas: &mut Vec<OutgoingTxMetadata>,
-    key_reader: impl Fn(&Keys) -> &Vec<K>,
-    transaction_to_bundle: impl Fn(&Transaction) -> Option<&B>,
-    bundle_to_output: impl Fn(&B) -> &L,
-    key_to_ivk: impl Fn(&K) -> Result<<D as Domain>::IncomingViewingKey, E>,
-    output_to_domain: impl Fn(&O) -> D,
-    memo_slice: impl Fn(&D::Memo) -> &[u8],
+    wallet_transactions: &Arc<RwLock<WalletTxns>>,
+    is_outgoing_transaction: &mut bool,
+    outgoing_metadatas: &mut Vec<OutgoingTxMetadata>,
 ) where
-    K: Clone,
-    for<'a> &'a L: IntoIterator<Item = &'a O>,
-    D: Domain,
-    O: ShieldedOutput<D, ENC_CIPHERTEXT_SIZE>,
+    D: zingo_traits::DomainWalletExt<Network>,
+    D::Note: Clone + PartialEq,
+    D::OutgoingViewingKey: std::fmt::Debug,
+    D::Recipient: zingo_traits::Recipient,
+    for<'a> &'a <<D as DomainWalletExt<Network>>::Bundle as zingo_traits::Bundle<D, Network>>::Spends:
+        IntoIterator<
+            Item = &'a <<D as DomainWalletExt<Network>>::Bundle as zingo_traits::Bundle<
+                D,
+                Network,
+            >>::Spend,
+        >,
+    for<'a> &'a <<D as DomainWalletExt<Network>>::Bundle as zingo_traits::Bundle<D, Network>>::Outputs:
+        IntoIterator<
+            Item = &'a <<D as DomainWalletExt<Network>>::Bundle as zingo_traits::Bundle<
+                D,
+                Network,
+            >>::Output,
+        >,
+    D::Memo: zingo_traits::ToBytes<512>,
 {
-    let local_keys = key_reader(&*keys.read().await).clone();
-    for output in transaction_to_bundle(transaction)
+    // Check if any of the nullifiers spent in this transaction are ours. We only need this for unconfirmed transactions,
```

Comment:
Does this language match our practice?   Do we say "nullifiers spent"? 

Or rather do we say that:

 `A spend of an output generates a nullifier, and a new output.`

?

Reply:
I think `nullifiers spent` is concise, and it would be nice to be more concise than having to say 'Check if any of the nullifiers in this transaction spend notes belonging to us', but 'nullifiers spent' doesn't convey that very well.